pkg/cloudevents/transport/httpb/transport.go

```diff
@@ -0,0 +1,327 @@
+func (t *Transport) Send(ctx context.Context, e event.Event) (context.Context, *event.Event, error) {
+	msg := binding.EventMessage(e)
+
+	if err := t.sender.Send(ctx, msg); err != nil {
+		return ctx, nil, err
+	}
+
+	return ctx, nil, nil
+}
+
+// SetReceiver implements Transport.SetReceiver
+func (t *Transport) SetReceiver(r transport.Receiver) {
+	t.consumer = r
+}
+
+// SetConverter implements Transport.SetConverter
+func (t *Transport) SetConverter(c transport.Converter) {
+	// TODO: implement converter
+	panic("not implemented")
+}
+
+// HasConverter implements Transport.HasConverter
+func (t *Transport) HasConverter() bool {
+	// TODO: implement converter
+	return false
+}
+
+// StartReceiver implements Transport.StartReceiver
+// NOTE: This is a blocking call.
+func (t *Transport) StartReceiver(ctx context.Context) error {
+	t.reMu.Lock()
+	defer t.reMu.Unlock()
+
+	logger := cecontext.LoggerFrom(ctx)
+
+	if t.receiver == nil {
+		t.receiver = http.NewReceiver()
+	}
+
+	if t.Handler == nil {
+		t.Handler = nethttp.NewServeMux()
+	}
+
+	if !t.handlerRegistered {
+		// handler.Handle might panic if the user tries to use the same path as the sdk.
+		t.Handler.Handle(t.GetPath(), t.receiver)
+		t.handlerRegistered = true
+	}
+
+	addr, err := t.listen()
+	if err != nil {
+		return err
+	}
+
+	t.server = &nethttp.Server{
+		Addr: addr.String(),
+		Handler: &ochttp.Handler{
+			Propagation: &tracecontext.HTTPFormat{},
+			// TODO: support middleware
+			//Handler:        attachMiddleware(t.Handler, t.middleware),
+			Handler:        t.Handler,
+			FormatSpanName: formatSpanName,
+		},
+	}
+
+	// Shutdown
+	defer func() {
+		_ = t.server.Close()
+		t.server = nil
+	}()
+
+	errChan := make(chan error, 1)
+	go func() {
+		errChan <- t.server.Serve(t.listener)
+	}()
+
+	go func() {
+		for {
+			msg, err := t.receiver.Receive(ctx)
+			if err == io.EOF {
+				// shut it down.
+				errChan <- err
+				break
+			} else if err != nil {
+				logger.Errorw("failed to receive:", zap.Error(err))
+				continue
+			}
+			e, _, err := binding.ToEvent(ctx, msg)
+
+			// TODO: response is not supported in PoC.
+			_, err = t.invokeReceiver(ctx, e)
+
+			err = msg.Finish(err)
+			if err != nil {
+				logger.Errorw("failed to Finish message: ", zap.Error(err))
+			}
+		}
+	}()
+
+	// wait for the server to return or ctx.Done().
+	select {
+	case <-ctx.Done():
+		// Try a gracefully shutdown.
+		timeout := DefaultShutdownTimeout
+		if t.ShutdownTimeout != nil {
+			timeout = *t.ShutdownTimeout
+		}
+		ctx, cancel := context.WithTimeout(context.Background(), timeout)
+		defer cancel()
+		err := t.server.Shutdown(ctx)
+		<-errChan // Wait for server goroutine to exit
+		return err
+	case err := <-errChan:
+		return err
+	}
+}
+
+// TODO: this is not what we want to use long term. For PoC.
+// Response is an http transport response.
+type Response struct {
+	StatusCode int
+	Message
+}
+
+// Message is an http transport message.
+type Message struct {
```

Comment:
Why do we need that? For the response?

Reply:
It is needed for the struct in Response but that section of the code is not used yet.